### PR TITLE
🍒 Python 3.12 Support

### PR DESCRIPTION
**Pull request by JDevlieghere**

### CODE REVIEW

**Comment by JDevlieghere** [comment]
@swift-ci test

**Comment by JDevlieghere** [comment]
@swift-ci test

File: lldb/unittests/ScriptInterpreter/Python/PythonTestSuite.cpp [hunk 1]
```diff
@@ -9,43 +9,27 @@
...
-  TestScriptInterpreterPython::Initialize();
-
   // Although we don't care about concurrency for the purposes of running
   // this test suite, Python requires the GIL to be locked even for
   // deallocating memory, which can happen when you call Py_DECREF or
   // Py_INCREF.  So acquire the GIL for the entire duration of this
   // test suite.
+  Py_InitializeEx(0);
   m_gil_state = PyGILState_Ensure();
+  PyRun_SimpleString("import sys");
```
Comment (adrian-prantl) [inline lldb/unittests/ScriptInterpreter/Python/PythonTestSuite.cpp:25]: Why is this there?

**Review by adrian-prantl — APPROVED** [review]
Conceptually looks good!

**Review by fredriss — APPROVED** [review]

File: lldb/unittests/ScriptInterpreter/Python/PythonTestSuite.cpp (hunk 1 above)
Comment (JDevlieghere) [reply lldb/unittests/ScriptInterpreter/Python/PythonTestSuite.cpp:25]: The Python Objects depend on the `sys` module to resolve a name in the context of a module (See `PythonObject::ResolveName` for details). Previously that was done behind the scenes in the call to Initialize.

**Comment by JDevlieghere** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test windows

**Pull request merged**